### PR TITLE
AddRegionsBoardPage

### PR DESCRIPTION
**Pull request by AlexKaschuk**
dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: EPlast/EPlast.BLL/Interfaces/GoverningBodies/IGoverningBodiesService.cs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+﻿using EPlast.BLL.DTO;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace EPlast.BLL.Interfaces.GoverningBodies
+{
+    public interface IGoverningBodiesService
+    {
+        Task<IEnumerable<OrganizationDTO>> GetOrganizationListAsync();
+
```
Comment (vovikboyko) [inline EPlast/EPlast.BLL/Interfaces/GoverningBodies/IGoverningBodiesService.cs:10]: empty row

**Review by Lunaflumen — APPROVED** [review]

File: EPlast/EPlast.BLL/Services/Region/RegionService.cs [hunk 2]
```diff
@@ -225,7 +228,7 @@ public async Task EndAdminsDueToDate()
         /// <inheritdoc />
         public async Task<IEnumerable<RegionForAdministrationDTO>> GetRegions()
         {
-            return _mapper.Map<IEnumerable<DataAccessRegion.Region>, IEnumerable<RegionForAdministrationDTO>>(await _repoWrapper.Region.GetAllAsync());
+            return _mapper.Map<IEnumerable<DataAccessRegion.Region>, IEnumerable<RegionForAdministrationDTO>>((await _repoWrapper.Region.GetAllAsync()).Where(r => r.Status != RegionsStatusType.RegionBoard));
         }
```
Comment (vovikboyko) [inline EPlast/EPlast.BLL/Services/Region/RegionService.cs:233]: empty row

File: EPlast/EPlast.DataAccess/Entities/Region/Region.cs [hunk 3]
```diff
@@ -25,5 +25,6 @@ public class Region
         public string HouseNumber { get; set; }
         public string OfficeNumber { get; set; }
         public int PostIndex { get; set; }
+        public RegionsStatusType Status { get; set; }
     }
 }
```
Comment (vovikboyko) [inline EPlast/EPlast.DataAccess/Entities/Region/Region.cs:30]: should add emty row

File: EPlast/EPlast.WebApi/Controllers/GoverningBodiesController.cs [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+        [HttpGet]
+        [Authorize(AuthenticationSchemes = "Bearer")]
+        public async Task<IActionResult> getOrganizations()
+        {
+            var governingBodies = await _governingBodiesService.GetOrganizationListAsync();
+            return Ok(governingBodies);
+        }
+
+
+
```
Comment (vovikboyko) [inline EPlast/EPlast.WebApi/Controllers/GoverningBodiesController.cs:35]: a lot of empty rows

File: EPlast/EPlast.WebApi/Controllers/RegionsController.cs [hunk 5]
```diff
@@ -32,6 +34,7 @@ public class RegionsController : ControllerBase
             IRegionAnnualReportService RegionAnnualReportService,
             UserManager<User> userManager,
             IDistributedCache cache)
+
```
Comment (vovikboyko) [inline EPlast/EPlast.WebApi/Controllers/RegionsController.cs:37]: empty row

File: EPlast/EPlast.WebApi/Controllers/RegionsController.cs [hunk 6]
```diff
@@ -159,8 +162,8 @@ public async Task<IActionResult> EditAdministrator(RegionAdministrationDTO admin
         public async Task<IActionResult> GetRegions(int page, int pageSize, string regionName)
         {
             string recordKey = "Regions_" + DateTime.Now.ToString("yyyyMMdd_hhmm");
-            IEnumerable<RegionDTO> regions = await _cache.GetRecordAsync<IEnumerable<RegionDTO>>(recordKey);
-            
+            //IEnumerable<RegionDTO> regions = await _cache.GetRecordAsync<IEnumerable<RegionDTO>>(recordKey);
```
Comment (vovikboyko) [inline EPlast/EPlast.WebApi/Controllers/RegionsController.cs:165]: vommented code

File: EPlast/EPlast.WebApi/Controllers/RegionsController.cs [hunk 7]
```diff
@@ -355,5 +358,18 @@ public async Task<IActionResult> GetAllRegionsReportsAsync()
...
+
+        /// <summary>
+        /// Method to get regions board
+        /// </summary>
+        /// <returns>region "Крайовий Провід Пласту"</returns>
+        /// <response code="200">Successful operation</response>
+        /// <response code="403">User hasn't access to region</response>
+        [HttpGet("regionsBoard")]
+        [Authorize(AuthenticationSchemes = "Bearer")]
+        public async Task<IActionResult> GetRegionsBoardAsync()
```
Comment (vovikboyko) [inline EPlast/EPlast.WebApi/Controllers/RegionsController.cs:370]: region or regionS? why plural?

File: EPlast/EPlast.WebApi/appsettings.json [hunk 8]
```diff
@@ -11,10 +11,10 @@
         }
     },
     "AllowedHosts": "*",
-    "ConnectionStrings": {
-        "EPlastDBConnection": "server=(localdb)\\mssqllocaldb;database=eplast;Trusted_Connection=true;",
-        "Redis": "10.244.0.96:6379"
-    },
+  "ConnectionStrings": {
+    "EPlastDBConnection": "Server=tcp:eplast.database.windows.net,1433;Initial Catalog=EPlast;Persist Security Info=False;User ID=Admin_EPlast;Password=7PJjTw#VuTZaD8G;MultipleActiveResultSets=False;Encrypt=True;TrustServerCertificate=False;Connection Timeout=30;",
```
Comment (vovikboyko) [inline EPlast/EPlast.WebApi/appsettings.json:15]: we shouldn't change this connection string. it gets from azure. here has to be local one

**Review by vovikboyko — CHANGES_REQUESTED** [review]

File: EPlast/EPlast.BLL/Services/Region/RegionService.cs [hunk 9]
```diff
@@ -210,13 +213,13 @@ public async Task EndAdminsDueToDate()
...
-            foreach(var admin in admins)
+            foreach (var admin in admins)
             {
-                if(DateTime.Compare((DateTime)admin.EndDate,DateTime.Now) < 0)
+                if (DateTime.Compare((DateTime)admin.EndDate, DateTime.Now) < 0)
                 {
                     admin.Status = false;
                     _repoWrapper.RegionAdministration.Update(admin);
-                   
+
```
Comment (vovikboyko) [inline EPlast/EPlast.BLL/Services/Region/RegionService.cs:222]: empty row

File: EPlast/EPlast.DataAccess/Entities/Region/Region.cs (hunk 3 above)
Comment (vovikboyko) [reply EPlast/EPlast.DataAccess/Entities/Region/Region.cs:30]: still should

**Review by vovikboyko — CHANGES_REQUESTED** [review]

File: EPlast/EPlast.WebApi/appsettings.json (hunk 8 above)
Comment (AlexKaschuk) [reply EPlast/EPlast.WebApi/appsettings.json]: done

File: EPlast/EPlast.WebApi/Controllers/RegionsController.cs (hunk 6 above)
Comment (AlexKaschuk) [reply EPlast/EPlast.WebApi/Controllers/RegionsController.cs]: done

File: EPlast/EPlast.WebApi/Controllers/GoverningBodiesController.cs (hunk 4 above)
Comment (AlexKaschuk) [reply EPlast/EPlast.WebApi/Controllers/GoverningBodiesController.cs]: done

File: EPlast/EPlast.DataAccess/Entities/Region/Region.cs (hunk 3 above)
Comment (AlexKaschuk) [reply EPlast/EPlast.DataAccess/Entities/Region/Region.cs:30]: done

File: EPlast/EPlast.BLL/Services/Region/RegionService.cs (hunk 2 above)
Comment (AlexKaschuk) [reply EPlast/EPlast.BLL/Services/Region/RegionService.cs]: done

File: EPlast/EPlast.BLL/Interfaces/GoverningBodies/IGoverningBodiesService.cs (hunk 1 above)
Comment (AlexKaschuk) [reply EPlast/EPlast.BLL/Interfaces/GoverningBodies/IGoverningBodiesService.cs]: done

**Review by vovikboyko — APPROVED** [review]

**Review by sooooolo — APPROVED** [review]

**Review by randomVovaKuhot — APPROVED** [review]

**Pull request merged**